Tests/SourceKitLSPTests/SwiftInterfaceTests.swift

```diff
@@ -319,6 +306,137 @@ final class SwiftInterfaceTests: XCTestCase {
     )
     XCTAssertEqual(diagnostics.fullReport?.items, [])
   }
+
+  func testInternalNavigationInSwiftInterface() async throws {
+    let testClient = try await TestSourceKitLSPClient(
+      capabilities: ClientCapabilities(experimental: [
+        GetReferenceDocumentRequest.method: .dictionary(["supported": .bool(true)])
+      ])
+    )
+    let uri = DocumentURI(for: .swiftinterface)
+
+    let positions = testClient.openDocument(
+      """
+      func test(x: 1️⃣String) {
+        let a: 2️⃣Int = 5
+      }
+      """,
+      uri: uri
+    )
+
+    // First, get definition for String to open the Swift interface
+    let stringDefinition = try await testClient.send(
+      DefinitionRequest(textDocument: TextDocumentIdentifier(uri), position: positions["1️⃣"])
+    )
+    let interfaceUri = try XCTUnwrap(stringDefinition?.locations?.only?.uri)
+    let interfaceContents = try await testClient.send(GetReferenceDocumentRequest(uri: interfaceUri))
+    // Open the interface document
+    testClient.send(
+      DidOpenTextDocumentNotification(
+        textDocument: TextDocumentItem(
+          uri: interfaceUri,
+          language: .swift,
+          version: 0,
+          text: interfaceContents.content
+        )
+      )
+    )
```

Comment:
I think you can simplify this slightly to 


```suggestion
    testClient.openDocument(interfaceContents.content, interfaceUri, language: .swift)
```